File: src/Goods.BackOffice/client-app/src/domain/settlements/settlementsTypes.ts
export enum SettlementsTypes {
	City = 1,
	Selo = 2,
	Derevnya = 3,
	PGT = 4
}

export namespace SettlementsTypes {
	export const getDisplayName = (type: SettlementsTypes): string => {
		switch (type) {
			case SettlementsTypes.City:
				return 'Город';
			case SettlementsTypes.Selo:
				return 'Село';
			case SettlementsTypes.Derevnya:
				return 'Деревня';
			case SettlementsTypes.PGT:
				return 'ПГТ';
		}
	};
}

File: src/Goods.Domain/Settlements/SettlementsTypes.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Security.Cryptography.X509Certificates;
using System.Text;
using System.Threading.Tasks;

public enum SettlementsTypes
{
    [Display(Name = "Город")]
    City = 1,

    [Display(Name = "Село")]
    Selo = 2,

    [Display(Name = "Деревня")]
    Derevnya = 3,

    [Display(Name = "ПГТ")]
    PGT = 4

}

public static class SettlementTypeExtention // Блок расширений для значений ENUM
{
    extension(SettlementsTypes type)
    {
        public Boolean CanHaveHistoricalValue =>
            type switch
            {
                SettlementsTypes.City => true,
                SettlementsTypes.Selo => true,
                SettlementsTypes.Derevnya => false,
                SettlementsTypes.PGT => true,
            };

        public Double ScorePoints =>
           type switch
           {
               SettlementsTypes.City => 1.3,
               SettlementsTypes.Selo => 0.5,
               SettlementsTypes.Derevnya => 0.8,
               SettlementsTypes.PGT => 1.0,
           };

    }
}